File: The queen on the chessboard.js
function availableMoves(position) {
    // Check if the input is a string
    if (typeof position !== 'string') {
        return [];
    }
    
    position = position.toUpperCase();
    
    // Check if the position is valid
    if (!/^[A-H][1-8]$/.test(position)) {
        return [];
    }
    
    const col = position.charCodeAt(0) - 'A'.charCodeAt(0);
    const row = parseInt(position[1]) - 1;
    
    const moves = [];
    
    // Vertical and horizontal moves
    for (let i = 0; i < 8; i++) {
        if (i !== row) {
            moves.push(position[0] + (i + 1));
        }
        if (i !== col) {
            moves.push(String.fromCharCode('A'.charCodeAt(0) + i) + position[1]);
        }
    }
    
    // Diagonal moves
    for (let i = 1; i < 8; i++) {
        if (col + i < 8 && row + i < 8) {
            moves.push(String.fromCharCode('A'.charCodeAt(0) + col + i) + (row + i + 1));
        }
        if (col - i >= 0 && row + i < 8) {
            moves.push(String.fromCharCode('A'.charCodeAt(0) + col - i) + (row + i + 1));
        }
        if (col + i < 8 && row - i >= 0) {
            moves.push(String.fromCharCode('A'.charCodeAt(0) + col + i) + (row - i + 1));
        }
        if (col - i >= 0 && row - i >= 0) {
            moves.push(String.fromCharCode('A'.charCodeAt(0) + col - i) + (row - i + 1));
        }
    }
    
    return moves.sort();
}

// Test cases
console.log(availableMoves("A1"));

File: The queen on the chessboard.py
def availableMoves(position):
    # Convert the position to uppercase string if it's not already
    if not isinstance(position, str):
        return []
    
    position = position.upper()
    
    # Check if the position is valid
    if len(position) != 2 or position[0] not in 'ABCDEFGH' or position[1] not in '12345678':
        return []
    
    col, row = ord(position[0]) - ord('A'), int(position[1]) - 1
    
    # generer les mouvement possible
    moves = []
    
    # Vertical et horizontal mouvenment
    for i in range(8):
        if i != row:
            moves.append(position[0] + str(i + 1))
        if i != col:
            moves.append(chr(i + ord('A')) + position[1])
    
    # Diagonal moves
    for i in range(1, 8):
        if 0 <= col + i < 8 and 0 <= row + i < 8:
            moves.append(chr(col + i + ord('A')) + str(row + i + 1))
        if 0 <= col - i < 8 and 0 <= row + i < 8:
            moves.append(chr(col - i + ord('A')) + str(row + i + 1))
        if 0 <= col + i < 8 and 0 <= row - i < 8:
            moves.append(chr(col + i + ord('A')) + str(row - i + 1))
        if 0 <= col - i < 8 and 0 <= row - i < 8:
            moves.append(chr(col - i + ord('A')) + str(row - i + 1))
    
    return sorted(moves)

# Test cases
print(availableMoves("A1"))
